mautic/mautic
# Change pre-commit hook to not allow a commit if there are errors in php-cs-fixer.

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ x ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]



#### Description:
Change the pre-commit hook, so it will not allow the commit if the committed code contains errors.
This will allow the developer to fix errors, and will not prevent commit by hunks.

NOTE: Changes are only for linux shell. I have no windows to test on.


#### Steps to test this PR:


1. Pull down for testing locally
2. Add some changes to code with errors
3. Try to commit
4. Observe the commit did not succeed.
5. Add changes that will pass the php-cs-fixer.
6. Add part of the changes that belong to *one file* to commit, leaving the file partially committed.
7. Observe only changes added to the commit are committed.
